pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -74,7 +74,10 @@ def done(self) -> bool:
         signals to stop (e.g. by early stopping).
         """
         max_steps_reached = self.max_steps is not None and self.global_step >= self.max_steps
-        return max_steps_reached or self.trainer.should_stop or self._num_training_batches_reached()
+        is_done = max_steps_reached or self._num_training_batches_reached()
+        # only when we are restarting we want to check whether the val loop has finished
+        val_loop_done = self.val_loop.done or not self.restarting
```

Comment:
the comment is ambiguous and does not match the logical implication that you implement below. 
so I would change to
```suggestion
        # when we are restarting we want to check whether the val loop has finished
        val_loop_done = not self.restarting or self.val_loop.done
```
because that's we mean in the context of this PR, right?